### PR TITLE
mouse absolute coordinate support

### LINKED ISSUE #20: Joystick Absolute Aim Mouse Emulation
Hello

Can QJoyPad do what I currently do with [Absolute Joystick Aim](https://github.com/mclang/GameControllerScripts/blob/master/Joystick%20Mouse%20Emulation/Absolute%20Joystick%20Aim.ahk) AutoHotkey script?

I use joystick to play Mechwarrior Online on Windows side because I like the more simulation feeling it gives. Problem is, MWO does not support joysticks properly, so the mouse emulation is necessary to be at least somewhat competitive against real mouse players.

When I heard that MWO might work under Linux using Steam Proton, I rejoiced. If I find a way to replicate the functionality of that AHK script in Linux side, I can finally stop booting into Windows completely!

Primary question is, does QJoyPad support absolute mouse move like what is used in the above AHK script?

If not, how could I possibly add it in QJoyPad, what source files to modify?

### PR DESCRIPTION
**Pull request by minger0**
Fixes #20

### CODE REVIEW

File: src/axis.h [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Axis : public QObject {
     Q_OBJECT
 
     //each axis can create a key press or move the mouse in one of four directions.
-    enum Interpretation { ZeroOne, Gradient, Absolute };
+    enum Interpretation { ZeroOne, Gradient, Absolute, Absolute2 = Absolute };
```
Comment (panzi) [inline src/axis.h:25]: What is this `Absolute2` good for?

File: src/event.cpp [hunk 2]
```diff
@@ -11,6 +11,19 @@ void sendevent(const FakeEvent &e) {
...
+      {
+        Screen* screen = XDefaultScreenOfDisplay(display);
+        static int rememberX = 0, rememberY = 0;
+        if (e.move.x) rememberX = e.move.x;
+        if (e.move.y) rememberY = e.move.y;
+        const int scaledX100 = rememberX * (XWidthOfScreen(screen)/2) / 100;
+	const int scaledY100 = rememberY * (XHeightOfScreen(screen)/2) / 100;
+	XTestFakeMotionEvent(display, DefaultScreen(display),
+                             XWidthOfScreen(screen)/2 + scaledX100, 
+                             XHeightOfScreen(screen)/2 + scaledY100, 0);
```
Comment (panzi) [inline src/event.cpp:24]: Indentation of these few lines is not right.

**Review by panzi — CHANGES_REQUESTED** [review]
It looks good, thank you! Just please take a look at the few things I've marked, then I'll merge this. :D

**Review by panzi — COMMENTED** [review]
What is this `Absolute2` good for?
PS: First time I use this review feature. Dunno if I could have done this in one review instead of two (one for each commit).

File: src/axis.h (hunk 1 above)
Comment (minger0) [reply src/axis.h:25]: there is a comment where I use it
there was a conflict with some external headers, so the name 'Absolute' is reserved

File: src/event.cpp (hunk 2 above)
Comment (minger0) [reply src/event.cpp:24]: not sure how you want to have it exactly, oneliner?
feel free to modify

**Comment by minger0** [comment]
> What is this `Absolute2` good for?
> PS: First time I use this review feature. Dunno if I could have done this in one review instead of two (one for each commit).
looks good to me

**Comment by panzi** [comment]
About indentation: Well, its a mess in this code base anyway. Still, at least keep the indentation the same (tabs Vs spaces) as the surrounding code. Kinda don't want to run a code formatter over it because it'll destroy git blame.

About `Absolute2`: Then please rename it `AbsolutePosition` or something like that. Although I think you could instead reference it with `Axis::Absolute` to resolve the name conflict at the site where you use that enum value. Definitely don't like an `Absolute2`.

Thank you for your contribution!

**Comment by minger0** [comment]
Axis::Absolute does not work against defines so I just took your advice and named it absolutepos 
+ addressed the disturbing tab/space
let me know

**Comment by panzi** [comment]
Thank you!!!!! :D

**Pull request merged**